src/lib.rs

```diff
@@ -327,6 +336,23 @@ fn do_camera_movement(
     *last_pos = Some(current_pos);
 }
 
+fn do_clamp_bounds(
+    mut query: Query<(&PanCam, &mut Transform, &OrthographicProjection)>,
+) {
```

Comment:
I wonder if it's strange that we only call this in response to user events, and not re-use it internally.

Reply:
There' a **small** amount of reuse in the sense that we call `clamp_to_safe_zone`, but I agree that it isn't ideal.

I made a judgment call that it could be wasteful to have to re-run through almost the entire code for the system that we've already done by the relevant points of both `do_camera_zoom` and `do_camera_movement`, re-extracting the same data that we'd already just calculated, just to squeeze out this little bit more reuse, especially since probably not **very** many library consumers will benefit from the new event.

I don't mind flipping that decision at all if you think it's not actually a big deal. Especially since I'm about to change this to a trigger/observer setup per the other comment, the downside will be even less.

Reply:
:page_facing_up: leaving this comment open for now; I don't mind changing the existing systems to use the new event if that's what you'd prefer, just let me know.